File: programmers/Level0/solution/_42_가장_큰_수_찾기/Solution.java
package programmers.Level0.solution._42_가장_큰_수_찾기;

import java.util.Arrays;

public class Solution {

    public static void main(String[] args) {
        int[] array = { 1, 8, 3 };

        Solution solution = new Solution();
        int[] result = solution.solution(array);
        System.out.println("result: " + Arrays.toString(result));
    }

    public int[] solution(int[] array) {
        int max = 0;
        int index = 0;
        for (int i = 0; i < array.length; i++) {
            if (max < array[i]) {
                max = array[i];
                index = i;
            }
        }

        int[] answer = new int[2];
        answer[0] = max;
        answer[1] = index;
        return answer;
    }
}

File: programmers/Level0/solution/_42_가장_큰_수_찾기/Solution.kt
package programmers.Level0.solution._42_가장_큰_수_찾기

fun main() {
    val array = intArrayOf(1, 8, 3)
    var result = solution(array)
    print("result: " + result.contentToString())

}

fun solution(array: IntArray): IntArray {
    var answer: IntArray = IntArray(2)
    for ((index, value) in array.withIndex()) {
        if (answer[0] < value) {
            answer[0] = value
            answer[1] = index
        }
    }
    return answer
}